mqt/qfr/qiskit/QuantumCircuit.hpp

```diff
@@ -192,6 +197,77 @@ namespace qc::qiskit {
             }
         }
 
+        static SymbolOrNumber parseSymbolicExpr(const py::object& pyExpr) {
+            static const std::regex summands("[+|-]?[^+-]+");
+            static const std::regex products("[\\*/]?[^\\*/]+");
+            // static const std::regex products(".*");
+            // static const products("")
+            auto exprStr = pyExpr.attr("__str__")().cast<std::string>();
+            exprStr.erase(std::remove(exprStr.begin(), exprStr.end(), ' '),
+                          exprStr.end()); // strip whitespace
+
+            auto sumIt  = std::sregex_iterator(exprStr.begin(), exprStr.end(), summands);
+            auto sumEnd = std::sregex_iterator();
+
+            qc::Symbolic sym;
+            bool         isConst = true;
+
+            while (sumIt != sumEnd) {
+                auto match    = *sumIt;
+                auto matchStr = match.str();
+                int  sign     = matchStr[0] == '-' ? -1 : 1;
```

Comment:
```suggestion
                const int  sign     = matchStr[0] == '-' ? -1 : 1;
```

Reply:
At this point it seems that const should just be the default.

Reply:
A short and nice article on that: https://belaycpp.com/2021/06/30/should-every-variable-be-const-by-default/